examples/benchmark/py/perf_run.py

```diff
@@ -0,0 +1,319 @@
+            end_time = timeit.default_timer()
+            meas_time = end_time - start_time
+            timings.append(meas_time)
+            print("Iterations {}: {:.6f} s".format(i, end_time - start_time))
+    
+    printStats("TensorRT", timings, precision)
+
+# Deploys inference run for different backend configurations
+def run(model, input_tensors, params, precision, is_trt_engine = False):
+    for backend in params.get('backend'):
+
+        if precision == 'int8':
+            if backend == 'all' or backend == 'torch':
+                print("int8 precision is not supported for torch runtime in this script yet")
+                return False
+
+            if backend == 'all' or backend == 'torch_tensorrt' or params.get('calibration_cache', None) == None:
+                print("int8 precision expects calibration cache file for inference")
+                return False
+
+        if backend == 'all':
+            run_torch(model, input_tensors, params, precision)
+            run_torch_tensorrt(model, input_tensors, params, precision)
+            run_tensorrt(model, input_tensors, params, precision, is_trt_engine)
+    
+        elif backend == "torch":
+            run_torch(model, input_tensors, params, precision)
+    
+        elif backend == "torch_tensorrt":
+            run_torch_tensorrt(model, input_tensors, params, precision)
+    
+        elif backend == "tensorrt":
+            run_tensorrt(model, input_tensors, params, precision, is_trt_engine)
+
+# Generate report
+def printStats(backend, timings, precision, batch_size = 1):
+    times = np.array(timings)
+    steps = len(times)
+    speeds = batch_size / times
+    time_mean = np.mean(times)
+    time_med = np.median(times)
+    time_99th = np.percentile(times, 99)
+    time_std = np.std(times, ddof=0)
+    speed_mean = np.mean(speeds)
+    speed_med = np.median(speeds)
+
+    msg = ("\n%s =================================\n"
+            "batch size=%d, num iterations=%d\n"
+            "  Median FPS: %.1f, mean: %.1f\n"
+            "  Median latency: %.6f, mean: %.6f, 99th_p: %.6f, std_dev: %.6f\n"
+            ) % (backend,
+                batch_size, steps,
+                speed_med, speed_mean,
+                time_med, time_mean, time_99th, time_std)
+    print(msg)
+    meas = {
+        'Backend' : backend,
+        'precision' : precision,
+        'Median(FPS)' : speed_med,
+        'Mean(FPS)' : speed_mean,
+        'Median-Latency(ms)' : time_med,
+        'Mean-Latency(ms)' : time_mean,
+        '99th_p' : time_99th,
+        'std_dev': time_std
+    }
+    results.append(meas)
+
+def precision_to_dtype(pr):
+    if pr == 'fp32':
+        return torch.float
+    elif pr == 'fp16' or pr == 'half':
+        return torch.half
+    else:
+        return torch.int8
+
+def load_model(params):
+    model = None
+    is_trt_engine = False
+    # Load torch model traced/scripted
+    model_file = os.path.join("models", params.get('model').get('filename'))
```

Comment:
Is there a reason we have to copy the model to `models` directory ? Like can we make it accept a path to a model wherever it is ?

Reply:
done